beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -394,6 +431,8 @@ where
         match rpc_event {
             RPCEvent::Request(id, req) => self.send_request(id, req),
             RPCEvent::Response(rpc_id, response) => {
+                // TODO: what happens here when shutting down?
```

Comment:
I guess let's be nice here. We would have sent a goodbye, but instead of just dropping an open stream, let's finish the last of our responses before terminating the connection entirely.
So we let this happen